qiskit/pulse/ir/ir.py

```diff
@@ -20,279 +20,107 @@
 """
 
 from __future__ import annotations
-from typing import List
-from abc import ABC, abstractmethod
 
-import numpy as np
+import rustworkx as rx
+from rustworkx import PyDAG
 
-from qiskit.pulse.exceptions import PulseError
+from qiskit.pulse.ir.alignments import Alignment
+from qiskit.pulse import Instruction
 
-from qiskit.pulse.transforms import AlignmentKind
-from qiskit.pulse.instructions import Instruction
+InNode = object()
+OutNode = object()
 
 
-class IrElement(ABC):
-    """Base class for Pulse IR elements"""
-
-    @property
-    @abstractmethod
-    def initial_time(self) -> int | None:
-        """Return the initial time of the element"""
-        pass
-
-    @property
-    @abstractmethod
-    def duration(self) -> int | None:
-        """Return the duration of the element"""
-        pass
-
-    @abstractmethod
-    def shift_initial_time(self, value: int):
-        """Shift ``initial_time``
-
-        Shifts ``initial_time`` to ``initial_time+value``.
-
-        Args:
-            value: The integer value by which ``initial_time`` is to be shifted.
-        """
-        pass
-
-    @property
-    def final_time(self) -> int | None:
-        """Return the final time of the element"""
-        try:
-            return self.initial_time + self.duration
-        except TypeError:
-            return None
-
-
-class IrInstruction(IrElement):
-    """Pulse IR Instruction
+class IrBlock:
```

Comment:
Can we give this a different name? I guess the original intention was "Intermediate representation of ScheduleBlock", and it makes me think this is still a collection of commands, i.e. [code block](https://en.wikipedia.org/wiki/Block_(programming)) + alignment. Since main purpose of this representation is sequencing, I prefer something like `SequenceIR` considering the lowering of the program; context alignment (lexical) -> sequence (ordered) -> schedule (t0 assigned).

Reply:
I don't have a strong preference here.